rest/src/main/java/org/jboss/pnc/rest/notifications/websockets/SessionBasedAttachedClient.java

```diff
@@ -39,8 +43,45 @@ public boolean isEnabled() {
     }
 
     @Override
-    public void sendMessage(Object messageBody) throws IOException {
-        session.getBasicRemote().sendText(outputConverter.apply(messageBody));
+    public void sendMessage(Object messageBody)
+            throws IOException, ExecutionException, InterruptedException, CancellationException {
+
+        // The sendText methos returns before the message is transmitted. The returned Future object is used to track the
+        // progress of the transmission. The Future's get method returns null upon successful completion. Errors in transmission
+        // are wrapped in the ExecutionException thrown when the Future object is queried.
+        Future<Void> future = session.getAsyncRemote().sendText(outputConverter.apply(messageBody));
+        
+        // wait for completion (forever, no timeout)
+        if (future.isDone()) {
+            future.get();
+        }
```

Comment:
Returning immediately is the purpose of AsyncRemote, you can provide SendHandler as 2nd parameter if you need a callback notification when sending completes. Which would be the right way to do it IMHO.
Current implementation is the same as if BasicRemote was used, which blocks the thread.